#### Importing the data and necessary libraries

In [2]:
import numpy as np
from sklearn.svm import SVR
exp_data = np.genfromtxt('input_data_train.csv', delimiter=',')
inp_train, label_train = exp_data[:,0], exp_data[:,1]

In [3]:
validation_data = np.genfromtxt('validation_data_test.csv', delimiter=',')
inp_test, label_test = validation_data[:,0], validation_data[:,1]

#### Fitting and Validating an SVR Model

In [4]:
clf = SVR(C=1.0, epsilon=0.2)
inp_train = inp_train.reshape(-1, 1)
inp_test = inp_test.reshape(-1, 1)
clf.fit(inp_train, label_train)
print("Residual sum of squares: %.2f" % np.mean((clf.predict(inp_test) - label_test) ** 2))
print("Coefficient of determination R^2 of the prediction: %.2f" % clf.score(inp_test, label_test))

Residual sum of squares: 3929.51
Coefficient of determination R^2 of the prediction: 0.84


#### Fitting and Validating a Linear Regression Model

In [5]:
from sklearn import datasets, linear_model
regr = linear_model.LinearRegression()
regr.fit(inp_train, label_train)
print("Residual sum of squares: %.2f" % np.mean((regr.predict(inp_test) - label_test) ** 2))
print("Coefficient of determination R^2 of the prediction: %.2f" % regr.score(inp_test, label_test))

Residual sum of squares: 2938.93
Coefficient of determination R^2 of the prediction: 0.88


#### Fitting a Polynomial Regression

In [6]:
from sklearn.linear_model import Ridge
from sklearn.preprocessing import PolynomialFeatures
from sklearn.pipeline import make_pipeline

degree = 2 # had the best results
polynomial_model = make_pipeline(PolynomialFeatures(degree), Ridge())
polynomial_model.fit(inp_train, label_train)
print("Residual sum of squares: %.2f" % np.mean((polynomial_model.predict(inp_test) - label_test) ** 2))
print("Coefficient of determination R^2 of the prediction: %.2f" % polynomial_model.score(inp_test, label_test))

Residual sum of squares: 2474.93
Coefficient of determination R^2 of the prediction: 0.90


#### Validating the Model with an Experiment running different utilizations for each core

In [37]:
final_validation = np.genfromtxt('final_validation_test.csv', delimiter=',')
inp_test1, label_test1 = final_validation[:,0], final_validation[:,1]
inp_test1 = inp_test1.reshape(-1, 1)
print("Residual sum of squares for regression: %.2f" % np.mean((regr.predict(inp_test1) - label_test1) ** 2))
print("Coefficient of determination R^2 of the prediction for regression: %.2f" % regr.score(inp_test1, label_test1))

Residual sum of squares for regression: 2467.31
Coefficient of determination R^2 of the prediction for regression: 0.76


In [39]:
print("Residual sum of squares for SVR: %.2f" % np.mean((clf.predict(inp_test1) - label_test1) ** 2))
print("Coefficient of determination R^2 of the prediction for SVR: %.2f" % clf.score(inp_test1, label_test1))

Residual sum of squares for SVR: 2112.20
Coefficient of determination R^2 of the prediction for SVR: 0.79


In [55]:
print("Residual sum of squares: %.2f" % np.mean((polynomial_model.predict(inp_test1) - label_test1) ** 2))
print("Coefficient of determination R^2 of the prediction: %.2f" % polynomial_model.score(inp_test1, label_test1))

Residual sum of squares: 1853.50
Coefficient of determination R^2 of the prediction: 0.82


#### Conclusion
The Polynomial Regression has had the best predictions. It was able to perform with .90% R^2 and Residual sum of squares of 2474.93 on the validation experiments with all 8 cores running under the same utilization. And it was able to predict a validation test data with .82 R^2 for an experiment with 8 cores running under different utilizations. 